cdap-ai-ext-vertexai/src/main/java/io/cdap/cdap/ai/ext/vertexai/VertexAIProvider.java

```diff
@@ -71,21 +72,22 @@ private Content createContent(String text) {
   /**
    * Summarizes the application details in the specified format using Vertex AI.
    *
-   * @param applicationDetail Details of the application to be summarized.
+   * @param appRequest Details of the application to be summarized.
    * @param format The format in which the summary should be provided.
    * @return This method will be returning a string which is summary in required format.
    */
   @Override
-  public String summarizeApp(ApplicationDetail applicationDetail, String format) {
+  public String summarizeApp(AppRequest appRequest, String format) {
     try {
-      String pipelineDetail = GSON.toJson(applicationDetail);
+      String pipelineDetail = GSON.toJson(appRequest);
       String prompt = conf.getPrompt().getPipelineMarkdownSummary();
-      String payload = pipelineDetail + "\n" + prompt;
+      String payload = new StringBuilder(pipelineDetail)
+          .append("\n").append(prompt).toString();
       GenerateContentResponse response = model.generateContent(createContent(payload));
       return ResponseHandler.getText(response);
     } catch (IOException ex) {
-      ex.printStackTrace();
-      return null;
+      LOG.error("Error summarizing application details", ex);
```

Comment:
Should be a warn

Reply:
Thrown directly to the method signature.